src/gf.c

```diff
@@ -2376,10 +2376,13 @@ JL_DLLEXPORT jl_function_t *jl_get_kwsorter(jl_value_t *ty)
     jl_methtable_t *mt = jl_argument_method_table(ty);
     if ((jl_value_t*)mt == jl_nothing)
         jl_error("cannot get keyword sorter for abstract type");
+    jl_datatype_t *dt = (jl_datatype_t*)jl_argument_datatype(ty);
+    assert(jl_is_datatype(dt));
```

Comment:
This assert is easily reachable, as it explicitly asserts that we do not have a unified method table (which we now do have)